libsolidity/analysis/ConstantEvaluator.cpp

```diff
@@ -41,22 +41,30 @@ void ConstantEvaluator::endVisit(BinaryOperation const& _operation)
 	auto right = type(_operation.rightExpression());
 	if (left && right)
 	{
-		auto commonType = left->binaryOperatorResult(_operation.getOperator(), right);
-		if (!commonType)
-			m_errorReporter.fatalTypeError(
-				_operation.location(),
-				"Operator " +
-				string(Token::toString(_operation.getOperator())) +
-				" not compatible with types " +
-				left->toString() +
-				" and " +
-				right->toString()
-			);
+		TypeResult result = left->binaryOperatorResult(_operation.getOperator(), right);
```

Comment:
We can still name the variable `commonType`

Reply:
Sure, this would make it more obvious that we're dealing with a type here. On the other hand, ``commonType.error()`` does look a bit weird to me, but maybe it's just my subjective perception and could be read as an actual type error. Either way, it's ok with me :)